pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -7838,72 +8174,65 @@ def solve_model(model, options=None):
+        if "gurobi_numeric_focus" in options.keys():
+            gurobi_numeric_focus = options["gurobi_numeric_focus"]
+        if "only_subsurface_block" in options.keys():
+            only_subsurface_block = options["only_subsurface_block"]
+
+    # Load solver
+    opt = get_solver(*solver) if type(solver) is tuple else get_solver(solver)
 
     # The below code is not the best way to check for solver but this works.
     # Checks for CPLEX using gams.
     if opt.options["solver"] == "CPLEX":
-        if options is not None:
-            if "running_time" in options.keys():
-                running_time = options["running_time"]
-            if "gap" in options.keys():
-                gap = options["gap"]
-
-            with open(f"{opt.options['solver']}.opt", "w") as f:
-                f.write(
-                    f"$onecho > {opt.options['solver']}.opt\n optcr={gap}\n running_time={running_time} $offecho"
-                )
-
-    # Checks for gurobi, gurobi direct, cbc.
-    elif opt.type == "gurobi_direct" or opt.type == "gurobi" or opt.type == "cbc":
-        if options is not None:
-            if "running_time" in options.keys():
-                running_time = options["running_time"]
-            if "gap" in options.keys():
-                gap = options["gap"]
-            if "deactivate_slacks" in options.keys():
-                deactivate_slacks = options["deactivate_slacks"]
-            if "scale_model" in options.keys():
-                # Note - we can't use "scale_model" as a local variable name because
-                # that is the name of the function that is called later to apply
-                # scaling to the model. So use "use_scaling" instead
-                use_scaling = options["scale_model"]
-            if "scaling_factor" in options.keys():
-                scaling_factor = options["scaling_factor"]
-            if "solver" in options.keys():
-                solver = options["solver"]
-            if "gurobi_numeric_focus" in options.keys():
-                gurobi_numeric_focus = options["gurobi_numeric_focus"]
-
-            if opt.type in ("gurobi_direct", "gurobi"):
-                # Apply Gurobi specific options
-                opt.options["mipgap"] = gap
-                opt.options["NumericFocus"] = gurobi_numeric_focus
-            elif opt.type in ("cbc"):
-                # Apply CBC specific option
-                opt.options["ratioGap"] = gap
-            # set maximum running time for solver
-            set_timeout(opt, timeout_s=running_time)
-
-    # deactivate slack variables if necessary
+        with open(f"{opt.options['solver']}.opt", "w") as f:
+            f.write(
+                f"$onecho > {opt.options['solver']}.opt\n optcr={gap}\n running_time={running_time} $offecho"
+            )
+
+    # Set maximum running time for solver
+    set_timeout(opt, timeout_s=running_time)
+
+    # Set solver gap
+    if opt.type in ("gurobi_direct", "gurobi"):
```

Comment:
Should this not be an elif statement? I agree it would not make a difference in ideal case but can cause problems if there are multiple solvers parsed.

Reply:
@JavalVyas2000 I don't think I am understanding your question here - can you clarify? This is an `if`/`elif` block that has a section for Gurobi and a section for CBC.

Reply:
I am saying it should be 
```
if CPLEX:
    Block of code
elif gurobi:
    Block of code
elif cbc:
    Block of code
else:
    Objective not supported. 
```

Here it is as follows:
```
if CPLEX:
    Block of code
if gurobi:
    Block of code
elif cbc:
    Block of code
else:
    Objective not supported. 
```

Reply:
I see. It would be fine with me if you want to try to reformat the code in that manner, and confirm that it works with CPLEX (I don't have CPLEX installed, so I wouldn't be able to test that part easily).

Reply:
Based on the offline discussion, this should not be a problem going forward.